addons/point_of_sale/static/src/app/services/pos_store.js

```diff
@@ -2173,10 +2173,11 @@ export class PosStore extends WithLazyGetterTrap {
         }
 
         const excludedProductIds = [
-            this.config.tip_product_id?.id,
-            ...this.hiddenProductIds,
-            ...this.session._pos_special_products_ids,
-        ];
+            this.config.tip_product_id?.product_tmpl_id.id,
+            ...this.session._pos_special_products_ids.map(
+                (id) => this.models["product.product"].get(id)?.product_tmpl_id.id
+            ),
+        ].filter(Boolean);
```

Comment:
The current failing tests in pos_loyalty are currently red because of this diff. That part will be handled in a separate PR in older version by @caburj